kolibri/content/content_db_router.py

```diff
@@ -40,18 +46,42 @@ def get_active_content_database(return_none_if_not_set=False):
     try:
         connections[alias]
     except ConnectionDoesNotExist:
+
+        # dynamically create a database entry to the desired content database
         filename = os.path.join(settings.CONTENT_DATABASE_DIR, alias + '.sqlite3')
         if not os.path.isfile(filename):
             raise KeyError("Content DB '%s' doesn't exist!!" % alias)
         connections.databases[alias] = {
             'ENGINE': 'django.db.backends.sqlite3',
             'NAME': filename,
         }
+
+        # check that the content database is not empty
         if not connections[alias].introspection.table_names():
             raise KeyError("Content DB '%s' is empty!!" % alias)
 
+    # if possible, attach the default database to the content database connection to enable joins
+    attach_default_database(alias)
+
     return alias
 
+def attach_default_database(alias):
+
+    # if the default database uses a sqlite file, attach it to the content database connection to enable joins
+    default_db = connections.databases[DEFAULT_DB_ALIAS]
+    if default_db["ENGINE"].endswith(".sqlite3") and default_db["NAME"].endswith(".sqlite3"):
+        default_db_path = connections.databases[DEFAULT_DB_ALIAS]["NAME"]
+        try:
+            # ensure we're connected to the content database before attaching the default database
+            if not connections[alias].connection:
+                connections[alias].connect()
+            # attach the default database to the connection
+            connections[alias].connection.execute("attach database '%s' as defaultdb;" % default_db_path)
+            # record the fact that the default database has been attached to this content database
+            _content_databases_with_attached_default_db.add(alias)
```

Comment:
You can, though here there's no need to.

The reason we do this here is to support the `default_database_is_attached` method, which gets called externally to determine whether it's attached. In the case of a non-sqlite default database, or an in-memory sqlite database, it won't attach it, and external code may need to know this and behave appropriately ([example](https://github.com/learningequality/kolibri/pull/357/files#diff-ebcc23f2084f43c36fed7b3d897da465R61)).